src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/DeferredBuildTest.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NUnit.Framework;
+using System.IO;
+using Xamarin.ProjectTools;
+
+namespace Xamarin.Android.Build.Tests
+{
+	[Parallelizable (ParallelScope.Children)]
+	public class DeferredBuildTest : BaseTest
+	{
+		[Test]
+		public void SelectivelyRunUpdateAndroidResources ()
+		{
+			var path = Path.Combine ("temp", TestName);
+			var app = new XamarinAndroidApplicationProject {
+				ProjectName = "MyApp",
+			};
+
+			app.SetProperty ("AndroidUseManagedDesignTimeResourceGenerator", "True");
+			app.SetProperty ("AndroidUseIntermediateDesignerFile", "True");
+
+			using (var appBuilder = CreateApkBuilder (Path.Combine (path, app.ProjectName))) {
+				appBuilder.Target = "Compile";
+				Assert.IsTrue (appBuilder.Build (app, parameters: new string[]{
+					// Normal DTB properties
+					"DesignTimeBuild=true",
+					"BuildingInsideVisualStudio=true",
+					"DeferredBuildSupported=true",
+					// The DTB passes down these two properties
+					"SkipCompilerExecution=true",
+					"ProvideCommandLineArgs=true",
+				}), "first app build should have succeeded.");
+
+				Assert.IsTrue (appBuilder.Output.IsTargetSkipped ("UpdateAndroidResources"), $"`UpdateAndroidResources` should be skipped for DTB when deferred build is supported!");
+
+				// The background build would run our UpdateAndroidResources in its BackgroundBuildDependsOn
+				appBuilder.Target = "UpdateAndroidResources";
+				Assert.IsTrue (appBuilder.Build (app, parameters: new string[]{
+					// Normal DTB properties
+					"DesignTimeBuild=true",
+					"BuildingInsideVisualStudio=true",
+					// BackgroundBuild targets set these two
+					"DeferredBuild=true",
+					"DeferredBuildSupported=true",
+					// The DTB passes down these two properties
+					"SkipCompilerExecution=true",
+					"ProvideCommandLineArgs=true",
+				}), "background build should have succeeded.");
+
+				Assert.IsFalse (appBuilder.Output.IsTargetSkipped ("UpdateAndroidResources"), $"`UpdateAndroidResources` should *not* be skipped in the deferred build!");
+
+				// Run the real build now
+				appBuilder.Target = "Build";
+				Assert.IsTrue (appBuilder.Build (app, parameters: new string[]{
+					"DesignTimeBuild=false",
+					"BuildingInsideVisualStudio=true",
+					"DeferredBuildSupported=true",
+				}), "real build should have succeeded.");
+			}
+		}
+
+		[Test]
+		public void RunUpdateAndroidResourcesIfBackgroundBuildNotSupported ()
+		{
+			var path = Path.Combine ("temp", TestName);
+			var app = new XamarinAndroidApplicationProject {
+				ProjectName = "MyApp",
+			};
+
+			app.SetProperty ("AndroidUseManagedDesignTimeResourceGenerator", "True");
+			app.SetProperty ("AndroidUseIntermediateDesignerFile", "True");
+
+			using (var appBuilder = CreateApkBuilder (Path.Combine (path, app.ProjectName))) {
+				appBuilder.Target = "Compile";
+				Assert.IsTrue (appBuilder.Build (app, parameters: new string[]{
```

Comment:
This should use the `DesignTimeBuild` method, I think?

Then you won't need to set all these properties--and if we need to add some of these properties to `DesignTimeBuild()` that would be fine.

Reply:
Not really. That method only sets `DesignTimeBuild=true`. I want to more closely model exactly what's called and what I see in the binlogs. Not sure if other tests want that too, maybe?